tools/federation/src/main/java/org/eclipse/rdf4j/federated/evaluation/TripleSource.java

```diff
@@ -82,6 +83,7 @@
 	 * @throws QueryEvaluationException
 	 */
 	public CloseableIteration<BindingSet, QueryEvaluationException> getStatements(String preparedQuery,
+			BindingSet queryBindings,
```

Comment:
Note that this is a slight change in the interface. Let me know if this is OK for a patch release.

I do consider this as an internal interface only.

As an alternitive I would need to duplicate the method (which is currently only called in a single location for single source queries)

Reply:
Usually for a public interface/class and a public method the method signature would have to stay the same except for between major version. In that case I would propose to introduce a new method, mark this one as deprecated and then we can remove it in 4.0.0 release. 

However, if it's an actual bug where the queryBindings should have been provided and are required for this method to function correctly, then I'm not sure.

@jeenbroekstra should decide if that is the case.

Reply:
I very much doubt that in practice, third parties will have started relying directly on this method. But of course, strictly speaking, this is a public method and therefore this is a breaking change. I'd rather err on the side of being overly strict than catching anyone who relies on us observing semver to be caught out. 

It would be fairly simple to keep the old method signature in deprecated form and fill in `null` for the BindingSet argument. Let's do that, just to be sure, and then we can comfortably get away with doing this as part of a patch.